conversion.c

```diff
@@ -0,0 +1,134 @@
+		bit = num % 2;
+		if (bit == cmp) revBin[i] = cmp + '0';
+		num /= 2;
+	}
+	char* bin = malloc (sizeof (char) * (len + 1));
+	if (bin == NULL) return NULL;
+	bin[len] = '\0';
+	for (int i = 0; i < len; i++) bin[i] = revBin[sIndex--];
+	free (revBin);
+	return bin;
+}
+
+/// <summary>Converts the given input to binary</summary>
+char* Binary (int num) {
+	int numCpy = num, power = 1, digit = 0;
+	bool largestNum = 0, isNegative = 0;                               // Variable largestNum sets if the number is of order (2**(8n)-1)                
+	if (num < 0) {
+		num = -num, numCpy = num, isNegative = 1;
+		int newNum;
+		for (digit = 0; numCpy > 0; digit++) numCpy = numCpy >> 1;
+		if (digit >= 31) return NULL;                                   // Only negative numbers upto 30 bits are evaluated 
+		for (int i = 0; i < digit + 1; i++) power = power << 1;
+		if (num == (power - 1)) {
+			largestNum = 1;
+			newNum = (num ^ ((2 * power) - 1));                          // 1's complement using EXOR bitwise operator 
+		} else newNum = (num ^ (power - 1));
+		newNum++;                                                       // 2's complement i.e incrementing 1's complement
+		num = newNum, numCpy = num;
+	}
+	for (digit = 0; numCpy > 0; digit++) numCpy = numCpy >> 1;
+	if (digit > 32) return NULL;
+	for (int i = 0; i < digit + 1; i++) power = power << 1;
+	if (num == (power - 1)) largestNum = 1;
+	char* bin = BinaryString (isNegative, NearestByte (digit, largestNum), num);
+	if (bin == NULL) exit (ERROR_MEM_ALLOC_FAILURE);
+	return bin;
+}
+
+/// <summary>Converts the given input to hexadecimal</summary>
+char* Hexadecimal (char* bin) {
+	int binLen = 0, hexLen = 0, length = 0, sum = 0, bit = 1;
+	if (bin == NULL) return NULL;
+	for (int i = 0; bin[i] != '\0'; i++) length++;
+	char* hexStr = malloc (sizeof (char) * ((length / 4) + 1));
+	if (hexStr == NULL) exit (ERROR_MEM_ALLOC_FAILURE);
+	while (binLen < length && hexLen < (length / 4)) {
+		sum = 0;
+		for (int i = 3; i >= 0; i--, binLen++) sum += (bin[binLen] - '0') * (bit << i);
+		hexStr[hexLen++] = Arr[sum];
+	}
+	hexStr[hexLen] = '\0';
+	return hexStr;
+}
+
+/// <summary>Method to print hexadecimal values using type specifier</summary>
+void Hex (int num) {
+	printf ("HEX: %X\n", num);
+}
+
+/// <summary>Prints the output of the program</summary>
+void Call (int num) {
+	char* bin = Binary (num), * hex = Hexadecimal (bin);
+	if (bin == NULL) printf ("The given negative number is too large(>= 31 bits)\n");
+	else printf ("BIN : %s\nHEX : %s\n", bin, hex);
+	free (bin);
+	free (hex);
+}
+
+int main () {
+	int c, lenIn = 0;
+	char inStr[12];
+	printf ("Enter an integer : ");
+	while ((c = getchar ()) != '\n') {
+		if ((lenIn == 0 && c == '-') || (c >= '0' && c <= '9' && lenIn <= 10)) inStr[lenIn++] = c; // First character can be negative, others within 0-9
+		else if (lenIn > 10) {                                                                     // Exceeds maximum bit length (32 bit)

```

Comment:
This type of error is called [integer overflow](https://en.wikipedia.org/wiki/Integer_overflow)
Good that you're checking.